### DISCUSSION TITLE
Problems with multi-running plugins

### DISCUSSION THREAD

**Discussion by jwgit3514** [appium/appium#18948] [Q&A]
Hello :)
I want to use images and appium-dashboard plugin together. However, when executed in the same way as the guide, both are not executed and seem to be recognized as one string.
Am I using it wrong?

PS > **appium --use-plugin=appium-dashboard,images**

[Appium] Welcome to Appium v2.0.1
[Appium] Non-default server args:
[Appium] {
[Appium]   usePlugins: [
[Appium]     **'appium-dashboard images'**
[Appium]   ]
[Appium] }
[Appium] Attempting to load driver uiautomator2...
[debug] [Appium] Requiring driver at C:\Users\jw3514\.appium\node_modules\appium-uiautomator2-driver
[Appium] Attempting to load driver xcuitest...
[debug] [Appium] Requiring driver at C:\Users\jw3514\.appium\node_modules\appium-xcuitest-driver
[Appium] Attempting to load driver espresso...
[debug] [Appium] Requiring driver at C:\Users\jw3514\.appium\node_modules\appium-espresso-driver
(node:7180) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 exit listeners added to [process]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
[Appium] Appium REST http interface listener started on http://0.0.0.0:4723
[Appium] You can provide the following URLS in your client code to connect to this server:
[Appium]        http://[my localhost]:4723/
[Appium]        http://127.0.0.1:4723/ (only accessible from the same host)
[Appium] Available drivers:
[Appium]   - uiautomator2@2.29.2 (automationName 'UiAutomator2')
[Appium]   - xcuitest@4.32.21 (automationName 'XCUITest')
[Appium]   - espresso@2.24.0 (automationName 'Espresso')
[Appium] Available plugins:
**[Appium]   - appium-dashboard@v2.0.2
[Appium]   - images@2.1.2
[Appium]   - device-farm@8.2.1
[Appium]   - element-wait@2.0.3**
[Appium] No plugins activated. Use the --use-plugins flag with names of plugins to activate
[Appium] Received SIGINT - shutting down
[debug] [AppiumDriver@fa5b] There are no active sessions for cleanup
[HTTP] Waiting until the server is closed
[HTTP] Received server close event

**Comment by saikrishna321** [comment]
@jwgit3514 you need to use "--use-plugins"

**Reply by jwgit3514** [reply]
Sorry
The example above is just one of several attempts.
The method you gave me has the same result.
Thank you for answer!

PS C:\Users\jw3514> appium server --use-plugins=appium-dashboard,images
[Appium] Welcome to Appium v2.0.1
[Appium] Non-default server args:
[Appium] {
[Appium]   usePlugins: [
[Appium]     'appium-dashboard images'
[Appium]   ]
[Appium] }
[Appium] Attempting to load driver uiautomator2...
[debug] [Appium] Requiring driver at C:\Users\jw3514\.appium\node_modules\appium-uiautomator2-driver
[Appium] Attempting to load driver xcuitest...
[debug] [Appium] Requiring driver at C:\Users\jw3514\.appium\node_modules\appium-xcuitest-driver
[Appium] Attempting to load driver espresso...
[debug] [Appium] Requiring driver at C:\Users\jw3514\.appium\node_modules\appium-espresso-driver
(node:30952) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 exit listeners added to [process]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
[Appium] Appium REST http interface listener started on http://0.0.0.0:4723
[Appium] You can provide the following URLS in your client code to connect to this server:
[Appium]        http://10.0.21.103:4723/
[Appium]        http://127.0.0.1:4723/ (only accessible from the same host)
[Appium] Available drivers:
[Appium]   - uiautomator2@2.29.2 (automationName 'UiAutomator2')
[Appium]   - xcuitest@4.32.21 (automationName 'XCUITest')
[Appium]   - espresso@2.24.0 (automationName 'Espresso')
[Appium] Available plugins:
[Appium]   - appium-dashboard@v2.0.2
[Appium]   - images@2.1.2
[Appium]   - device-farm@8.2.1
[Appium]   - element-wait@2.0.3
[Appium] No plugins activated. Use the --use-plugins flag with names of plugins to activate

**Comment by jlipps** [comment]
I cannot reproduce this on macos. Maybe it's a windows shell parsing issue (perhaps a result of [this](https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-arrays?view=powershell-7.3#other-syntax). Maybe wrap it in quotes:

```
appium server --use-plugins="appium-dashboard,images"
```

**Reply by jwgit3514** [reply]
I think I've tried every phrase searchable on the internet. If it's an OS problem, I'll have to run the server separately. Thank you for answer!

**Reply by jlipps** [reply]
Did running it with quotes help?

**Reply by jwgit3514** [reply]
I got the same result :,(

appium server -ka 800 --base-path=/wd/hub --use-plugins="images,appium-dashboard"
[Appium] Welcome to Appium v2.0.1
[Appium] Non-default server args:
[Appium] {
[Appium]   basePath: '/wd/hub',
[Appium]   keepAliveTimeout: 800,
[Appium]   usePlugins: [
[Appium]     'images appium-dashboard'
[Appium]   ]
[Appium] }
[Appium] Attempting to load driver uiautomator2...
[debug] [Appium] Requiring driver at C:\Users\jw3514\.appium\node_modules\appium-uiautomator2-driver
[Appium] Attempting to load driver xcuitest...
[debug] [Appium] Requiring driver at C:\Users\jw3514\.appium\node_modules\appium-xcuitest-driver
[Appium] Attempting to load driver espresso...
[debug] [Appium] Requiring driver at C:\Users\jw3514\.appium\node_modules\appium-espresso-driver
(node:30132) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 exit listeners added to [process]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
[Appium] Appium REST http interface listener started on http://0.0.0.0:4723/wd/hub
[Appium] You can provide the following URLS in your client code to connect to this server:
[Appium]        http://10.0.21.103:4723/wd/hub
[Appium]        http://127.0.0.1:4723/wd/hub (only accessible from the same host)
[Appium] Available drivers:
[Appium]   - uiautomator2@2.29.2 (automationName 'UiAutomator2')
[Appium]   - xcuitest@4.32.21 (automationName 'XCUITest')
[Appium]   - espresso@2.24.0 (automationName 'Espresso')
[Appium] Available plugins:
[Appium]   - appium-dashboard@v2.0.2
[Appium]   - images@2.1.2
[Appium]   - device-farm@8.2.1
[Appium]   - element-wait@2.0.3
[Appium] No plugins activated. Use the --use-plugins flag with names of plugins to activate

**Reply by jwgit3514** [reply]
Other attempts, such as separating with an extra '/', gave the same result.

**Comment by jwgit3514** [comment]
i found the cause
Plugin options are not entered normally in powershell.
If you run appium server in cmd, it works normally.
Thanks to both of you for your replies

**Reply by jlipps** [reply]
Yep, there's probably a way to do it in powershell. I don't use windows or powershell though so I'm not sure what it would be.
